pkg/iac/terraform/state/backend/http_reader.go

```diff
@@ -27,25 +31,23 @@ func NewHTTPReader(rawURL string, opts *Options) (*HTTPBackend, error) {
 		req.Header.Add(key, value)
 	}
 
-	client := &http.Client{}
 	res, err := client.Do(req)
 	if err != nil {
 		return nil, err
 	}
 
-	buf := new(bytes.Buffer)
-	_, err = buf.ReadFrom(res.Body)
+	body, err := io.ReadAll(res.Body)
```

Comment:
`client.Do` and response read should be done in the `Read` func IMHO, it's kinda unexpected to have this kind of logic within a `New...` method.